lighthouse-core/scripts/legacy-javascript/run.js

```diff
@@ -8,11 +8,25 @@
 /* eslint-disable no-console */
 
 const fs = require('fs');
+const path = require('path');
 const glob = require('glob');
 const {execFileSync} = require('child_process');
+const crypto = require('crypto');
 const LegacyJavascript = require('../../audits/legacy-javascript.js');
 const networkRecordsToDevtoolsLog = require('../../test/network-records-to-devtools-log.js');
-const VARIANT_DIR = `${__dirname}/variants`;
+
+// Create variants in a directory named-cached by contents of this script and the lockfile.
+// This folder is in the CI cache, so that the time consuming part of this test only runs if
+// the output would change.
+removeCoreJs(); // (in case the script was canceled halfway - there shouldn't be a core-js dep checked in.)
```

Comment:
it does seem a little odd to control this here (both running here and not in `main()` and `yarn` commands inside this file in general), but I haven't reviewed the full control flow so I'll assume it makes sense :)